ams/api/src/main/java/com/netease/arctic/ams/api/process/TableProcess.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.ams.api.process;
+
+import com.netease.arctic.ams.api.TableRuntime;
+
+/**
+ * An abstract table process to handle table state.
+ *
+ * @param <T>
+ */
+public abstract class TableProcess<T extends TableState> implements AmoroProcess<T> {
+
+  protected final T state;
+  protected final TableRuntime tableRuntime;
+  private final SimpleFuture submitFuture = new SimpleFuture();
+  private final SimpleFuture completeFuture = new SimpleFuture();
+  private volatile ProcessStatus status = ProcessStatus.RUNNING;
```

Comment:
Why do we need to store the status and failed reason outside as well, if it's already included in the state?

Reply:
Process keeps the future result and State keeps persistent (when we call setStatus() in State, that means a persistency. When a future has a result, status in Process would be changed).

This is a consideration for result delivering. The other difference is that status is process is always idempotent, which however in State, we shall use StateField to keep status consistent with other fields.